mithril-stm/src/schnorr_signature/verification_key.rs

```diff
@@ -0,0 +1,140 @@
+#[derive(Debug, Clone, Copy, Default, PartialEq, Eq)]
+pub struct SchnorrVerificationKey(pub(crate) JubjubSubgroup);
+
+impl SchnorrVerificationKey {
+    /// Convert a `SchnorrVerificationKey` into bytes.
+    pub fn to_bytes(self) -> [u8; 32] {
+        self.0.to_bytes()
+    }
+
+    /// Convert bytes into a `SchnorrVerificationKey`.
+    ///
+    /// The bytes must represent a Jubjub Subgroup point or the conversion will fail
+    pub fn from_bytes(bytes: &[u8]) -> StmResult<Self> {
+        let mut verification_key_bytes: [u8; 32] = [0u8; 32];
+        verification_key_bytes.copy_from_slice(
+            bytes
+                .get(0..32)
+                .ok_or(anyhow!(SchnorrSignatureError::SerializationError))?,
+        );
+        let point = JubjubSubgroup::from_bytes(&verification_key_bytes)
+            .into_option()
+            .ok_or(anyhow!(SchnorrSignatureError::SerializationError))?;
+
+        Ok(SchnorrVerificationKey(point))
+    }
+}
+
+impl From<&crate::SchnorrSigningKey> for SchnorrVerificationKey {
+    /// Convert a Schnorr secret key into a verification key
+    ///
+    /// This is done by computing `vk = g * sk` where g is the generator
+    /// of the subgroup and sk is the schnorr secret key
+    fn from(signing_key: &crate::SchnorrSigningKey) -> Self {
+        let generator = JubjubSubgroup::generator();
+
+        SchnorrVerificationKey(generator * signing_key.0)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use dusk_jubjub::Fq as JubjubBase;
+    use dusk_jubjub::SubgroupPoint as JubjubSubgroup;
+    use ff::Field;
+    use group::Group;
+    use rand_chacha::ChaCha20Rng;
+    use rand_core::SeedableRng;
+
+    use crate::schnorr_signature::{SchnorrSigningKey, SchnorrVerificationKey};
+
+    #[test]
+    fn generate_verification_key() {
+        let mut rng = ChaCha20Rng::from_seed([0u8; 32]);
+        let sk = SchnorrSigningKey::generate(&mut rng).unwrap();
+        let g = JubjubSubgroup::generator();
+        let vk = g * sk.0;
+
+        let vk_from_sk = SchnorrVerificationKey::from(&sk);
+
+        assert_eq!(vk, vk_from_sk.0);
+    }
+
+    #[test]
+    fn verify_fail_verification_key_not_on_curve() {
+        let msg = vec![0, 0, 0, 1];
+        let seed = [0u8; 32];
+        let mut rng = ChaCha20Rng::from_seed(seed);
+        let sk = SchnorrSigningKey::generate(&mut rng).unwrap();
+        let vk1 = SchnorrVerificationKey::from(&sk);
+        let sig = sk.sign(&msg, &mut rng).unwrap();
+        let vk2 = SchnorrVerificationKey(JubjubSubgroup::from_raw_unchecked(
+            JubjubBase::ONE,
+            JubjubBase::ONE,
+        ));
+
+        let result1 = sig.verify(&msg, &vk1);
+        let result2 = sig.verify(&msg, &vk2);
+
+        assert!(
+            result1.is_ok(),
+            "Correct verification key used, test should pass."
+        );
+        assert!(
+            result2.is_err(),
+            "Invalid verification key used, test should fail."
+        );
```

Comment:
Same here with an `expect_err` (here and everywhere else this appears in the PR):
```suggestion
        result2.expect_err("Invalid verification key used, test should fail.");
```